iceoryx_hoofs/vocabulary/include/iox/string.hpp

```diff
@@ -26,46 +26,53 @@
 #include <algorithm>
 #include <cstdint>
 #include <cstring>
-#include <iostream>
 
 namespace iox
 {
 namespace log
 {
 class LogStream;
 }
+
+/// @brief This helper struct can be used to enable the comparison methods for a custom string type by specializing for
+/// a custom 'T'
+/// @note The custom string needs a specialisation of 'GetData' and 'GetSize' as well
+template <typename T>
+struct is_custom_string : public std::false_type
+{
+};
+
 // AXIVION DISABLE STYLE AutosarC++19_03-A18.1.1 : C-array type usage is intentional
 
 template <typename T, typename ReturnType>
 using IsStringOrCharArrayOrChar =
-    typename std::enable_if<((is_cxx_string<T>::value || is_char_array<T>::value)
-                             || (std::is_same<T, std::string>::value || std::is_same<T, char>::value)),
+    typename std::enable_if<(is_cxx_string<T>::value || is_char_array<T>::value || std::is_same<T, char>::value
```

Comment:
Since you changed `IsCxxStringOrCharArray` to `IsIoxStringOrCharArray` would it make sense to change `is_cxx_string` to `is_iox_string`?